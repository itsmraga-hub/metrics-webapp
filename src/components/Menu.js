import { useSelector } from 'react-redux';

import style from './css/Menu.module.css';

const Menu = () => {
  const { all } = useSelector((state) => state.cases);

  const totalConfirmed = all.reduce((sum, country) => sum += country.confirmed, 0);
  const population = all.reduce((sum, country) => {
    if (country.population === undefined || null) {
      sum += 0;
      return sum;
    }
    sum += country.population;
    return sum;
  }, 0);

  const deaths = all.reduce((sum, country) => sum += country.deaths, 0);
  return (
    <section className={style.Menu}>
      <h2>All Time Stats</h2>
      <div>
        <p>
          {totalConfirmed}
          <span>Total confirmed cases</span>
        </p>
        <p>
          {population}
          <span>Population</span>
        </p>
        <p>
          {deaths}
          <span>Total Deaths</span>
        </p>
      </div>
    </section>
  );
};

export default Menu;
