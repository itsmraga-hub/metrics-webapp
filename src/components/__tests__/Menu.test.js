import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';

import Menu from '../Menu';

it('<Menu /> Renders Properly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Menu />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
