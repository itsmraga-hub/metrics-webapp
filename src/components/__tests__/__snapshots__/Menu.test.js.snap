// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Renders Properly 1`] = `
<section
  className="Menu"
>
  <h2>
    All Time Stats
  </h2>
  <div>
    <p>
      0
      <span>
        Total confirmed cases
      </span>
    </p>
    <p>
      0
      <span>
        Population
      </span>
    </p>
    <p>
      0
      <span>
        Total Deaths
      </span>
    </p>
  </div>
</section>
`;
